docs/conventions/component.md

```diff
@@ -0,0 +1,461 @@
+
+#### Generic mixin use
+
+Call of the mixin defined in generic component, with custom parameters.
+
+#### Additional css rules
+
+If needed, some extra css rules could be added after to specialize the component.
+
+Example:
+```scss
+// Import base and generic
+@import '@ecl/ec-base';
+@import '@ecl/generic-component-navigation-menu/generic-component-navigation-menu';
+
+// Check if overridden dependencies are already loaded, if needed
+@include check-imports(('ec-component-link', 'ec-component-button'));
+
+// Use generic mixin
+@include exports('ec-component-navigation-menu') {
+  @include ecl-navigation-menu(
+    $bar-bg-mobile: map-get($ecl-colors, 'grey-15'),
+    $bar-bg-desktop: map-get($ecl-colors, 'grey-10'),
+    $toggle-btn-color: map-get($ecl-colors, 'blue-100'),
+    $group-bg: map-get($ecl-colors, 'grey-5'),
+    $separator-mobile: map-get($ecl-colors, 'grey-15'),
+    $link-color: map-get($ecl-colors, 'blue-100'),
+    $active-color: map-get($ecl-colors, 'blue-100'),
+    $active-bg: map-get($ecl-colors, 'grey-10')
+  );
+
+  // Add custom rules
+  .ecl-navigation-menu__item--active {
+    [...]
+  }
+  .ecl-navigation-menu__group::before {
+    [...]
+  }
+}
+```
+
+### Specific case: editor preset
+
+If the component alters a default html tag display (like links, blockquotes, ...), it should also provide css rules for editing.
+
+**Please note that this is a specific case, most components don't have to create editor rules**
+
+To add editor rules, there should be some extra rules (prefixed by `.ecl-editor`), which override default html tag.
+This rules should be put in a mixin called `ecl-editor-{component name}`, in a separated file called `{component name}--editor.scss`.
+
+Example:
+```scss
+/* File path: generic-component-link--editor.scss */
+
+@mixin ecl-editor-generic-component-link() {
+  .ecl-editor a {
+    box-sizing: border-box;
+    color: $ecl-color-primary;
+    margin: 0;
+    text-decoration: underline;
+    [...]
+  }
+}
+```
+
+## Twig
+
+Every component should have a directly usable twig file. The goal is to be able to include the component in other molecules/organisms (by putting most of the content in context).
+
+Apart from [rules that should be applied to all twig files](twig), there are some specifications for components:
+
+### File name
+
+Twig file name should be the same as component's name.
+
+Example:
+```
+generic-component-navigation-menu.twig
+```
+
+### Generic component
+
+Generic component twig file defines all the markup for the component. It should contains all the logic required to use the component, as it is intended to be used by systems components.
+
+This twig file consists in different sections (see link above for more details):
+* Parameters/Blocks: list of exposed parameters and blocks
+* Internal properties: definition of custom variables
+* Internal logic: preprocess and other custom logic
+* Print: html markup
+
+Example:
+```twig
+{#
```

Comment:
The `blockquote` is not too big of a component and may be small enough to demonstrate the concept pefectly.